tests/test_connection.py

```diff
@@ -2757,10 +2757,36 @@ def test_channel_opener(node_factory):
 
     # close and check for 'closer'
     l1.rpc.close(l2.rpc.getinfo()["id"])
+    l1.daemon.wait_for_log("State changed from CHANNELD_NORMAL to")
+    l2.daemon.wait_for_log("State changed from CHANNELD_NORMAL to")
+
+    # 'closer' should now be set accordingly
     assert(l1.rpc.listpeers()['peers'][0]['channels'][0]['closer'] == 'local')
     assert(l2.rpc.listpeers()['peers'][0]['channels'][0]['closer'] == 'remote')
 
 
+def test_channel_closer_unilateral(node_factory, bitcoind):
+    """ Check 'closer' attribute on `listpeers` channels for unilateral close """
+    l1, l2 = node_factory.line_graph(2)
+    scid = l1.get_channel_scid(l2)
+
+    # close channel unilaterally by stopping l1 first
+    l1.rpc.stop()
+    wait_for(lambda: not only_one(l2.rpc.listpeers()['peers'])['connected'])
+    l2.rpc.close(scid, 1)           # force close after 1sec timeout
+
+    # restart and wait
+    l1.restart()
+    bitcoind.generate_block(1)      # gets force close confirmed
+    l1.daemon.wait_for_log("State changed from CHANNELD_NORMAL to")
+
+    # Check for 'closer':
+    #  - l2 has closer 'local' because he forced closed it
+    #  - l1 has closer 'remote' because he just saw the onchain tx
+    assert(l1.rpc.listpeers()['peers'][0]['channels'][0]['closer'] == 'remote')
+    assert(l2.rpc.listpeers()['peers'][0]['channels'][0]['closer'] == 'local')
```

Comment:
A faster/more CI friendly thing to do would be to find a test where we *already* do a unilateral close and add this check to that.

Reply:
Thx.
Did so and also integrated the `test_channel_opener` into the `state_change` tests...